### DISCUSSION TITLE
401 unauthorized error when starting the safe exam browser

### DISCUSSION THREAD

**Discussion by jorritpr** [SafeExamBrowser/seb-win-refactoring#500] [Q&A] [answered]
Hi,

We're trying to set up a test environment for the safe exam browser and SEB server but we can't get it to work properly. 

We've set up a SEB server in kubernetes (https://sebserver-test.ifed.rug.nl) and in the seb server we've configured an institution, connection configuration, exam configuration, configuration template, LMS setup, exam template and we've activated an exam (a demo quiz). 

When I export the connection configuration and start this I get a "configuration successful" notification, so it seems okay up until that point. When starting the safe exam browser I get a 401 unauthorized message straightaway, saying ""an error occurred while trying to communicate with the SEB server. (see the attached screenshot)

As far as I know we haven't configured any passwords and I can't figure out what causes this error message. Also we don't see any incoming network traffic to the seb server when starting the SEB browser.

When I click on fallback, the SEB browser starts correctly with the fallback url we've configured in the connection configuration. 

![image](https://user-images.githubusercontent.com/104197354/199206465-fcf170ec-4baf-4d1a-b823-7f15cf5400f5.png)

Could anyone tell me what configuration setting we've missed and point me in the right direction? Thanks in advance! 

kind regards,
Jorrit

**Comment by anhefti** [comment, marked as answer]
Hi Jorrit

Can you please check if the respective Connection Configuration that you downloaded and use, is still active on the SEB Server.
If so, and since this is a Kubernetes setup, you have to make sure also the webservices are available from the internet and not blocked by some proxy or load balancer. See the following picture for a better understanding of the Kubernetes setup:

![image](https://user-images.githubusercontent.com/39078777/199209044-9531bfed-226e-4a0c-ad60-61f20684326d.png)

Note also that you have to use sticky-sessions on IP level for the SEB - webservice connection

If everything above seems to work correctly, please send us also the Connection Configuration you use and the SEB Client Logs, so we can analyze them more deeply. 

best regards
Andreas

**Reply by jorritpr** [reply]
Hi Andreas, thanks for your reply. 

We've made the webservice available and that did the trick! 

kind regards,
Jorrit
